From 7eb30b345e12e6b81747fe4337a22367216430a5 Mon Sep 17 00:00:00 2001
From: sys_icgvert <icgvert@intel.com>
Date: Tue, 26 Apr 2016 19:01:03 -0700
Subject: [PATCH] atomisp2: css2400 and 2401 irci_ecr-master_20160426_2016

Merge CSS2.1 for css2400 and 2401_legacy pkg.

Change-Id: I919e44db7c502301b54b512780f920b14b546eeb
FW: tag irci_ecr-master_20160426_2016
Original-Tracked_On: #H1504238773
Signed-off-by: sys_icgvert <icgvert@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/116716
Approver: WeiX Meng <weix.meng@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Signed-off-by: Yang Fan F <fan.f.yang@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/118582
Build: Dessert M BTNS (SC)
Build: Dessert M BXT RVP (SC)
Build: Dessert M CHT FFD (SC)
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-31237
Signed-off-by: Tao Wei <wei.tao@intel.com>
Reviewed-on: https://android.intel.com:443/503925
---
 .../pci/atomisp2/css/css_2400_system/spmem_dump.c  | 210 ++++++++++-----------
 .../css/css_2401_csi2p_system/spmem_dump.c         | 166 ++++++++--------
 .../pci/atomisp2/css/css_2401_system/spmem_dump.c  | 206 ++++++++++----------
 .../media/pci/atomisp2/css/css_api_version.h       |   5 +-
 .../camera/drivers/media/pci/atomisp2/css/ia_css.h |   2 +-
 .../drivers/media/pci/atomisp2/css/ia_css_types.h  |   3 +-
 .../media/pci/atomisp2/css/ia_css_version_data.h   |   2 +-
 .../pci/atomisp2/css/runtime/queue/src/queue.c     |  14 +-
 .../media/pci/atomisp2/css/sh_css_firmware.c       |   4 +-
 .../drivers/media/pci/atomisp2/css/sh_css_params.c |   1 +
 .../drivers/media/pci/atomisp2/css/sh_css_params.h |   1 +
 11 files changed, 314 insertions(+), 300 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2400_system/spmem_dump.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2400_system/spmem_dump.c
index 97d1f69..95e7a93 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2400_system/spmem_dump.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2400_system/spmem_dump.c
@@ -23,7 +23,7 @@ more details.
 
 /* function input_system_acquisition_stop: AD8 */
 
-/* function longjmp: 6BF2 */
+/* function longjmp: 6C11 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_HIVE_IF_SRST_MASK
@@ -37,9 +37,9 @@ more details.
 #define HIVE_ADDR_sp_HIVE_IF_SRST_MASK 0x1C8
 #define HIVE_SIZE_sp_HIVE_IF_SRST_MASK 16
 
-/* function tmpmem_init_dmem: 68EC */
+/* function tmpmem_init_dmem: 690B */
 
-/* function ia_css_isys_sp_token_map_receive_ack: 6230 */
+/* function ia_css_isys_sp_token_map_receive_ack: 624F */
 
 /* function ia_css_dmaproxy_sp_set_addr_B: 371A */
 
@@ -65,9 +65,9 @@ more details.
 
 /* function ia_css_queue_get_size: 4D35 */
 
-/* function ia_css_queue_load: 533C */
+/* function ia_css_queue_load: 535B */
 
-/* function setjmp: 6BFB */
+/* function setjmp: 6C1A */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_sp2host_isys_event_queue
@@ -81,9 +81,9 @@ more details.
 #define HIVE_ADDR_sp_sem_for_sp2host_isys_event_queue 0x48E4
 #define HIVE_SIZE_sp_sem_for_sp2host_isys_event_queue 20
 
-/* function ia_css_dmaproxy_sp_wait_for_ack: 717C */
+/* function ia_css_dmaproxy_sp_wait_for_ack: 719B */
 
-/* function ia_css_sp_rawcopy_func: 5567 */
+/* function ia_css_sp_rawcopy_func: 5586 */
 
 /* function ia_css_tagger_buf_sp_pop_marked: 2D93 */
 
@@ -111,9 +111,9 @@ more details.
 #define HIVE_ADDR_sp_vbuf_raw 0x314
 #define HIVE_SIZE_sp_vbuf_raw 4
 
-/* function ia_css_sp_bin_copy_func: 548D */
+/* function ia_css_sp_bin_copy_func: 54AC */
 
-/* function ia_css_queue_item_store: 509A */
+/* function ia_css_queue_item_store: 50B9 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_pipe_private_metadata_bufs
@@ -211,11 +211,11 @@ more details.
 #define HIVE_ADDR_sp_vbuf_handles 0x659C
 #define HIVE_SIZE_sp_vbuf_handles 960
 
-/* function ia_css_queue_store: 5200 */
+/* function ia_css_queue_store: 521F */
 
 /* function ia_css_sp_flash_register: 2FC8 */
 
-/* function ia_css_isys_sp_backend_create: 5EA3 */
+/* function ia_css_isys_sp_backend_create: 5EC2 */
 
 /* function ia_css_pipeline_sp_init: 187F */
 
@@ -285,7 +285,7 @@ more details.
 #define HIVE_ADDR_sp_host_sp_com 0x42FC
 #define HIVE_SIZE_sp_host_sp_com 220
 
-/* function ia_css_queue_get_free_space: 4E5F */
+/* function ia_css_queue_get_free_space: 4E61 */
 
 /* function exec_image_pipe: 658 */
 
@@ -301,7 +301,7 @@ more details.
 #define HIVE_ADDR_sp_sp_init_dmem_data 0x6524
 #define HIVE_SIZE_sp_sp_init_dmem_data 24
 
-/* function ia_css_sp_metadata_start: 5C80 */
+/* function ia_css_sp_metadata_start: 5C9F */
 
 /* function ia_css_bufq_sp_init_buffer_queues: 3037 */
 
@@ -329,7 +329,7 @@ more details.
 #define HIVE_ADDR_sp_sp_per_frame_data 0x43D8
 #define HIVE_SIZE_sp_sp_per_frame_data 4
 
-/* function ia_css_rmgr_sp_vbuf_dequeue: 6640 */
+/* function ia_css_rmgr_sp_vbuf_dequeue: 665F */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_host2sp_psys_event_queue_handle
@@ -355,11 +355,11 @@ more details.
 #define HIVE_ADDR_sp_xmem_bin_addr 0x43DC
 #define HIVE_SIZE_sp_xmem_bin_addr 4
 
-/* function tmr_clock_init: 690C */
+/* function tmr_clock_init: 692B */
 
 /* function ia_css_pipeline_sp_run: 1426 */
 
-/* function memcpy: 6C9B */
+/* function memcpy: 6CBA */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_GP_DEVICE_BASE
@@ -387,13 +387,13 @@ more details.
 
 /* function input_system_reg_store: B18 */
 
-/* function ia_css_isys_sp_frontend_start: 60B9 */
+/* function ia_css_isys_sp_frontend_start: 60D8 */
 
-/* function ia_css_uds_sp_scale_params: 69A4 */
+/* function ia_css_uds_sp_scale_params: 69C3 */
 
 /* function ia_css_circbuf_increase_size: E35 */
 
-/* function __divu: 6C19 */
+/* function __divu: 6C38 */
 
 /* function ia_css_thread_sp_get_state: C78 */
 
@@ -425,9 +425,9 @@ more details.
 
 /* function ia_css_parambuf_sp_handle_parameter_sets: 1281 */
 
-/* function ia_css_spctrl_sp_set_state: 5CAF */
+/* function ia_css_spctrl_sp_set_state: 5CCE */
 
-/* function ia_css_thread_sem_sp_signal: 6E9D */
+/* function ia_css_thread_sem_sp_signal: 6EBC */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_IRQ_BASE
@@ -453,13 +453,13 @@ more details.
 #define HIVE_ADDR_sp_TIMED_CTRL_BASE 0x40
 #define HIVE_SIZE_sp_TIMED_CTRL_BASE 4
 
-/* function ia_css_isys_sp_isr: 736A */
+/* function ia_css_isys_sp_isr: 7389 */
 
-/* function ia_css_isys_sp_generate_exp_id: 6451 */
+/* function ia_css_isys_sp_generate_exp_id: 6470 */
 
-/* function ia_css_rmgr_sp_init: 653B */
+/* function ia_css_rmgr_sp_init: 655A */
 
-/* function ia_css_thread_sem_sp_init: 6F6C */
+/* function ia_css_thread_sem_sp_init: 6F8B */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_reading_cb_frame
@@ -487,7 +487,7 @@ more details.
 
 /* function ia_css_dmaproxy_sp_execute: 35F0 */
 
-/* function ia_css_queue_is_empty: 72C9 */
+/* function ia_css_queue_is_empty: 72E8 */
 
 /* function ia_css_pipeline_sp_has_stopped: 1858 */
 
@@ -507,13 +507,13 @@ more details.
 #define HIVE_ADDR_sp_current_sp_thread 0x1DC
 #define HIVE_SIZE_sp_current_sp_thread 4
 
-/* function ia_css_spctrl_sp_get_spid: 5CB6 */
+/* function ia_css_spctrl_sp_get_spid: 5CD5 */
 
 /* function ia_css_bufq_sp_reset_buffers: 30BE */
 
-/* function ia_css_dmaproxy_sp_read_byte_addr: 71AA */
+/* function ia_css_dmaproxy_sp_read_byte_addr: 71C9 */
 
-/* function ia_css_rmgr_sp_uninit: 6534 */
+/* function ia_css_rmgr_sp_uninit: 6553 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_threads_stack
@@ -531,7 +531,7 @@ more details.
 
 /* function ia_css_parambuf_sp_wait_for_in_param: 1048 */
 
-/* function ia_css_isys_sp_token_map_get_exp_id: 6319 */
+/* function ia_css_isys_sp_token_map_get_exp_id: 6338 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_all_cb_elems_param
@@ -583,7 +583,7 @@ more details.
 #define HIVE_ADDR_sp_sem_for_reading_cb_param 0x4970
 #define HIVE_SIZE_sp_sem_for_reading_cb_param 40
 
-/* function ia_css_queue_get_used_space: 4E13 */
+/* function ia_css_queue_get_used_space: 4E15 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_cont_capt_start
@@ -609,7 +609,7 @@ more details.
 #define HIVE_ADDR_sp_tmp_heap 0x6288
 #define HIVE_SIZE_sp_tmp_heap 640
 
-/* function ia_css_rmgr_sp_get_num_vbuf: 6844 */
+/* function ia_css_rmgr_sp_get_num_vbuf: 6863 */
 
 /* function ia_css_ispctrl_sp_output_compute_dma_info: 438D */
 
@@ -627,11 +627,11 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_s3a_bufs 0x4E00
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_s3a_bufs 60
 
-/* function ia_css_queue_is_full: 4EAA */
+/* function ia_css_queue_is_full: 4EAC */
 
 /* function debug_buffer_init_isp: E4 */
 
-/* function ia_css_isys_sp_frontend_uninit: 6073 */
+/* function ia_css_isys_sp_frontend_uninit: 6092 */
 
 /* function ia_css_tagger_sp_exp_id_is_locked: 206D */
 
@@ -647,7 +647,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_rmgr_sp_mipi_frame_sem 0x695C
 #define HIVE_SIZE_sp_ia_css_rmgr_sp_mipi_frame_sem 60
 
-/* function ia_css_rmgr_sp_refcount_dump: 661B */
+/* function ia_css_rmgr_sp_refcount_dump: 663A */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_pipe_private_isp_parameters_id
@@ -687,7 +687,7 @@ more details.
 #define HIVE_ADDR_sp_host2sp_isys_event_queue_handle 0x4E50
 #define HIVE_SIZE_sp_host2sp_isys_event_queue_handle 12
 
-/* function ia_css_thread_sp_yield: 6E1B */
+/* function ia_css_thread_sp_yield: 6E3A */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_all_cbs_param_desc
@@ -719,7 +719,7 @@ more details.
 
 /* function ia_css_dmaproxy_sp_vmem_read: 3590 */
 
-/* function ia_css_ifmtr_sp_init: 64A2 */
+/* function ia_css_ifmtr_sp_init: 64C1 */
 
 /* function initialize_sp_group: 668 */
 
@@ -727,9 +727,9 @@ more details.
 
 /* function ia_css_thread_sp_init: D31 */
 
-/* function ia_css_isys_sp_reset_exp_id: 6449 */
+/* function ia_css_isys_sp_reset_exp_id: 6468 */
 
-/* function qos_scheduler_update_fps: 6994 */
+/* function qos_scheduler_update_fps: 69B3 */
 
 /* function ia_css_ispctrl_sp_set_stream_base_addr: 4A81 */
 
@@ -825,13 +825,13 @@ more details.
 
 /* function encode_and_post_sp_event_non_blocking: A0C */
 
-/* function ia_css_isys_sp_frontend_destroy: 614B */
+/* function ia_css_isys_sp_frontend_destroy: 616A */
 
 /* function is_ddr_debug_buffer_full: 2CC */
 
-/* function ia_css_isys_sp_frontend_stop: 608B */
+/* function ia_css_isys_sp_frontend_stop: 60AA */
 
-/* function ia_css_isys_sp_token_map_init: 63E7 */
+/* function ia_css_isys_sp_token_map_init: 6406 */
 
 /* function ia_css_tagger_buf_sp_get_oldest_marked_offset: 2D05 */
 
@@ -851,9 +851,9 @@ more details.
 
 /* function debug_enqueue_ddr: EE */
 
-/* function ia_css_rmgr_sp_refcount_init_vbuf: 65D6 */
+/* function ia_css_rmgr_sp_refcount_init_vbuf: 65F5 */
 
-/* function dmaproxy_sp_read_write: 7248 */
+/* function dmaproxy_sp_read_write: 7267 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_dmaproxy_isp_dma_cmd_buffer
@@ -891,17 +891,17 @@ more details.
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_service 0x32C0
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_service 4
 
-/* function ia_css_dmaproxy_sp_process: 6F94 */
+/* function ia_css_dmaproxy_sp_process: 6FB3 */
 
 /* function ia_css_tagger_buf_sp_mark_from_end: 2F8D */
 
-/* function ia_css_isys_sp_backend_rcv_acquire_ack: 5D58 */
+/* function ia_css_isys_sp_backend_rcv_acquire_ack: 5D77 */
 
-/* function ia_css_isys_sp_backend_pre_acquire_request: 5D6E */
+/* function ia_css_isys_sp_backend_pre_acquire_request: 5D8D */
 
 /* function ia_css_ispctrl_sp_init_cs: 3A4F */
 
-/* function ia_css_spctrl_sp_init: 5CC4 */
+/* function ia_css_spctrl_sp_init: 5CE3 */
 
 /* function sp_event_proxy_init: 6C4 */
 
@@ -967,7 +967,7 @@ more details.
 
 /* function sp_dma_proxy_reset_channels: 3875 */
 
-/* function ia_css_isys_sp_backend_acquire: 5E79 */
+/* function ia_css_isys_sp_backend_acquire: 5E98 */
 
 /* function ia_css_tagger_sp_update_size: 2C17 */
 
@@ -1027,11 +1027,11 @@ more details.
 
 /* function ia_css_dmaproxy_sp_channel_acquire: 38A1 */
 
-/* function ia_css_rmgr_sp_add_num_vbuf: 6820 */
+/* function ia_css_rmgr_sp_add_num_vbuf: 683F */
 
 /* function ia_css_tagger_frame_in_capture_pipe: 2019 */
 
-/* function ia_css_isys_sp_token_map_create: 6430 */
+/* function ia_css_isys_sp_token_map_create: 644F */
 
 /* function __ia_css_dmaproxy_sp_wait_for_ack_text: 355C */
 
@@ -1067,7 +1067,7 @@ more details.
 
 /* function ia_css_tagger_sp_create: 2BC5 */
 
-/* function tmpmem_acquire_dmem: 68CD */
+/* function tmpmem_acquire_dmem: 68EC */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_MMU_BASE
@@ -1131,17 +1131,17 @@ more details.
 
 /* function sp_isys_copy_request: 6A8 */
 
-/* function ia_css_rmgr_sp_refcount_retain_vbuf: 66B0 */
+/* function ia_css_rmgr_sp_refcount_retain_vbuf: 66CF */
 
 /* function ia_css_thread_sp_set_priority: CC3 */
 
 /* function sizeof_hmem: BF6 */
 
-/* function tmpmem_release_dmem: 68BC */
+/* function tmpmem_release_dmem: 68DB */
 
 /* function cnd_input_system_cfg: 392 */
 
-/* function __ia_css_sp_rawcopy_func_critical: 72F3 */
+/* function __ia_css_sp_rawcopy_func_critical: 7312 */
 
 /* function __ia_css_dmaproxy_sp_process_text: 3500 */
 
@@ -1153,7 +1153,7 @@ more details.
 
 /* function ia_css_tagger_buf_sp_pop_unmarked_and_unlocked: 2D47 */
 
-/* function __modu: 6C5F */
+/* function __modu: 6C7E */
 
 /* function ia_css_dmaproxy_sp_init_isp_vector: 3562 */
 
@@ -1173,11 +1173,11 @@ more details.
 #define HIVE_ADDR_sp_GDC_BASE 0x44
 #define HIVE_SIZE_sp_GDC_BASE 8
 
-/* function ia_css_queue_local_init: 5074 */
+/* function ia_css_queue_local_init: 5093 */
 
-/* function sp_event_proxy_callout_func: 6D2C */
+/* function sp_event_proxy_callout_func: 6D4B */
 
-/* function qos_scheduler_schedule_stage: 6940 */
+/* function qos_scheduler_schedule_stage: 695F */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_thread_sp_num_ready_threads
@@ -1205,13 +1205,13 @@ more details.
 
 /* function ia_css_ispctrl_sp_isp_done_row_striping: 4373 */
 
-/* function __ia_css_isys_sp_isr_text: 6175 */
+/* function __ia_css_isys_sp_isr_text: 6194 */
 
-/* function ia_css_queue_dequeue: 4EF2 */
+/* function ia_css_queue_dequeue: 4EF9 */
 
-/* function is_qos_standalone_mode: 691B */
+/* function is_qos_standalone_mode: 693A */
 
-/* function ia_css_dmaproxy_sp_configure_channel: 71C1 */
+/* function ia_css_dmaproxy_sp_configure_channel: 71E0 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_current_thread_fiber_sp
@@ -1227,7 +1227,7 @@ more details.
 
 /* function ia_css_circbuf_pop: FCD */
 
-/* function memset: 6CDE */
+/* function memset: 6CFD */
 
 /* function irq_raise_set_token: B6 */
 
@@ -1259,7 +1259,7 @@ more details.
 #define HIVE_ADDR_sp_isp_ph 0x6558
 #define HIVE_SIZE_sp_isp_ph 28
 
-/* function ia_css_isys_sp_token_map_flush: 6375 */
+/* function ia_css_isys_sp_token_map_flush: 6394 */
 
 /* function ia_css_ispctrl_sp_init_ds: 3BDB */
 
@@ -1293,17 +1293,17 @@ more details.
 
 /* function __ia_css_dmaproxy_sp_configure_channel_text: 36D1 */
 
-/* function ia_css_framebuf_sp_wait_for_in_frame: 684B */
+/* function ia_css_framebuf_sp_wait_for_in_frame: 686A */
 
-/* function ia_css_sp_rawcopy_tag_frame: 59D5 */
+/* function ia_css_sp_rawcopy_tag_frame: 59F4 */
 
 /* function isp_hmem_clear: B1F */
 
-/* function ia_css_framebuf_sp_release_in_frame: 688E */
+/* function ia_css_framebuf_sp_release_in_frame: 68AD */
 
-/* function ia_css_isys_sp_backend_snd_acquire_request: 5DCB */
+/* function ia_css_isys_sp_backend_snd_acquire_request: 5DEA */
 
-/* function ia_css_isys_sp_token_map_is_full: 61FC */
+/* function ia_css_isys_sp_token_map_is_full: 621B */
 
 /* function input_system_acquisition_run: AF3 */
 
@@ -1335,7 +1335,7 @@ more details.
 #define HIVE_ADDR_sp_isp_pool 0x308
 #define HIVE_SIZE_sp_isp_pool 4
 
-/* function ia_css_rmgr_sp_rel_gen: 657D */
+/* function ia_css_rmgr_sp_rel_gen: 659C */
 
 /* function ia_css_tagger_sp_unblock_clients: 2A31 */
 
@@ -1353,15 +1353,15 @@ more details.
 #define HIVE_ADDR_sp_event_any_pending_mask 0x320
 #define HIVE_SIZE_sp_event_any_pending_mask 8
 
-/* function ia_css_isys_sp_backend_push: 5D82 */
+/* function ia_css_isys_sp_backend_push: 5DA1 */
 
 /* function sh_css_decode_tag_descr: 352 */
 
 /* function debug_enqueue_isp: 27B */
 
-/* function qos_scheduler_update_stage_budget: 6923 */
+/* function qos_scheduler_update_stage_budget: 6942 */
 
-/* function ia_css_spctrl_sp_uninit: 5CBD */
+/* function ia_css_spctrl_sp_uninit: 5CDC */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_HIVE_IF_SWITCH_CODE
@@ -1445,11 +1445,11 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_exp_ids 0x5C08
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_exp_ids 70
 
-/* function ia_css_queue_item_load: 5166 */
+/* function ia_css_queue_item_load: 5185 */
 
-/* function ia_css_spctrl_sp_get_state: 5CA8 */
+/* function ia_css_spctrl_sp_get_state: 5CC7 */
 
-/* function ia_css_isys_sp_token_map_uninit: 6392 */
+/* function ia_css_isys_sp_token_map_uninit: 63B1 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_callout_sp_thread
@@ -1477,7 +1477,7 @@ more details.
 #define HIVE_ADDR_sp_SP_PMEM_BASE 0x0
 #define HIVE_SIZE_sp_SP_PMEM_BASE 4
 
-/* function ia_css_isys_sp_token_map_snd_acquire_req: 6302 */
+/* function ia_css_isys_sp_token_map_snd_acquire_req: 6321 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_isp_input_stream_format
@@ -1491,19 +1491,19 @@ more details.
 #define HIVE_ADDR_sp_sp_isp_input_stream_format 0x42E0
 #define HIVE_SIZE_sp_sp_isp_input_stream_format 20
 
-/* function __mod: 6C4B */
+/* function __mod: 6C6A */
 
 /* function ia_css_dmaproxy_sp_init_dmem_channel: 3620 */
 
 /* function ia_css_thread_sp_join: CF4 */
 
-/* function ia_css_dmaproxy_sp_add_command: 72B3 */
+/* function ia_css_dmaproxy_sp_add_command: 72D2 */
 
-/* function ia_css_sp_metadata_thread_func: 5B80 */
+/* function ia_css_sp_metadata_thread_func: 5B9F */
 
-/* function __sp_event_proxy_func_critical: 6D19 */
+/* function __sp_event_proxy_func_critical: 6D38 */
 
-/* function ia_css_sp_metadata_wait: 5C6F */
+/* function ia_css_sp_metadata_wait: 5C8E */
 
 /* function ia_css_circbuf_peek_from_start: EFD */
 
@@ -1513,9 +1513,9 @@ more details.
 
 /* function sp_isys_copy_func: 68A */
 
-/* function ia_css_isys_sp_backend_flush: 5DEB */
+/* function ia_css_isys_sp_backend_flush: 5E0A */
 
-/* function ia_css_isys_sp_backend_frame_exists: 5D07 */
+/* function ia_css_isys_sp_backend_frame_exists: 5D26 */
 
 /* function ia_css_sp_isp_param_init_isp_memories: 4C19 */
 
@@ -1551,7 +1551,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_isys_sp_frontend_states 0x653C
 #define HIVE_SIZE_sp_ia_css_isys_sp_frontend_states 12
 
-/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 7193 */
+/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 71B2 */
 
 /* function ia_css_ispctrl_sp_done_ds: 3BC2 */
 
@@ -1597,11 +1597,11 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_frames 0x5C50
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_frames 280
 
-/* function qos_scheduler_init_stage_budget: 6981 */
+/* function qos_scheduler_init_stage_budget: 69A0 */
 
-/* function ia_css_isys_sp_backend_release: 5E60 */
+/* function ia_css_isys_sp_backend_release: 5E7F */
 
-/* function ia_css_isys_sp_backend_destroy: 5E8A */
+/* function ia_css_isys_sp_backend_destroy: 5EA9 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp2host_buffer_queue_handle
@@ -1615,15 +1615,15 @@ more details.
 #define HIVE_ADDR_sp_sp2host_buffer_queue_handle 0x5D68
 #define HIVE_SIZE_sp_sp2host_buffer_queue_handle 96
 
-/* function ia_css_isys_sp_token_map_check_mipi_frame_size: 62C6 */
+/* function ia_css_isys_sp_token_map_check_mipi_frame_size: 62E5 */
 
 /* function ia_css_ispctrl_sp_init_isp_vars: 48E6 */
 
-/* function ia_css_isys_sp_frontend_has_empty_mipi_buffer_cb: 5EDC */
+/* function ia_css_isys_sp_frontend_has_empty_mipi_buffer_cb: 5EFB */
 
 /* function sp_warning: 870 */
 
-/* function ia_css_rmgr_sp_vbuf_enqueue: 6670 */
+/* function ia_css_rmgr_sp_vbuf_enqueue: 668F */
 
 /* function ia_css_tagger_sp_tag_exp_id: 21C7 */
 
@@ -1655,7 +1655,7 @@ more details.
 #define HIVE_ADDR_sp_sp_isp_addresses 0x61BC
 #define HIVE_SIZE_sp_sp_isp_addresses 172
 
-/* function ia_css_rmgr_sp_acq_gen: 6595 */
+/* function ia_css_rmgr_sp_acq_gen: 65B4 */
 
 /* function receiver_reg_load: ACA */
 
@@ -1683,7 +1683,7 @@ more details.
 #define HIVE_ADDR_sp_host_sp_queues_initialized 0x42F4
 #define HIVE_SIZE_sp_host_sp_queues_initialized 4
 
-/* function ia_css_queue_uninit: 5032 */
+/* function ia_css_queue_uninit: 5051 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_ispctrl_sp_isp_started
@@ -1729,7 +1729,7 @@ more details.
 #define HIVE_ADDR_sp_sp_metadata_thread 0x4C10
 #define HIVE_SIZE_sp_sp_metadata_thread 72
 
-/* function ia_css_queue_enqueue: 4F7C */
+/* function ia_css_queue_enqueue: 4F8F */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_flash_sp_request
@@ -1757,7 +1757,7 @@ more details.
 #define HIVE_ADDR_sp_tagger_frames 0x4C60
 #define HIVE_SIZE_sp_tagger_frames 168
 
-/* function ia_css_isys_sp_token_map_snd_capture_req: 6324 */
+/* function ia_css_isys_sp_token_map_snd_capture_req: 6343 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_reading_if
@@ -1775,11 +1775,11 @@ more details.
 
 /* function ia_css_pipeline_sp_start: 186A */
 
-/* function ia_css_thread_default_callout: 6E14 */
+/* function ia_css_thread_default_callout: 6E33 */
 
-/* function ia_css_sp_rawcopy_init: 554F */
+/* function ia_css_sp_rawcopy_init: 556E */
 
-/* function tmr_clock_read: 6902 */
+/* function tmr_clock_read: 6921 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ISP_BAMEM_BASE
@@ -1793,7 +1793,7 @@ more details.
 #define HIVE_ADDR_sp_ISP_BAMEM_BASE 0x318
 #define HIVE_SIZE_sp_ISP_BAMEM_BASE 4
 
-/* function ia_css_isys_sp_frontend_rcv_capture_ack: 5F8B */
+/* function ia_css_isys_sp_frontend_rcv_capture_ack: 5FAA */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_sems_for_sp2host_buf_queues
@@ -1859,11 +1859,11 @@ more details.
 
 /* function ia_css_tagger_buf_sp_pop_from_start: 2CC9 */
 
-/* function __div: 6C03 */
+/* function __div: 6C22 */
 
-/* function ia_css_isys_sp_frontend_create: 615C */
+/* function ia_css_isys_sp_frontend_create: 617B */
 
-/* function ia_css_rmgr_sp_refcount_release_vbuf: 668F */
+/* function ia_css_rmgr_sp_refcount_release_vbuf: 66AE */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_flash_sp_in_use
@@ -1877,7 +1877,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_use 0x4D10
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_use 4
 
-/* function ia_css_thread_sem_sp_wait: 6EE8 */
+/* function ia_css_thread_sem_sp_wait: 6F07 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_sleep_mode
@@ -1907,7 +1907,7 @@ more details.
 #define HIVE_ADDR_sp_sp_max_cb_elems 0x148
 #define HIVE_SIZE_sp_sp_max_cb_elems 8
 
-/* function ia_css_queue_remote_init: 5054 */
+/* function ia_css_queue_remote_init: 5073 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_isp_stop_req
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/spmem_dump.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/spmem_dump.c
index 76ca3df..09d167d 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/spmem_dump.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_csi2p_system/spmem_dump.c
@@ -21,9 +21,9 @@ more details.
 
 #define _hrt_cell_load_program_sp(proc) _hrt_cell_load_program_embedded(proc, sp)
 
-/* function longjmp: 6C37 */
+/* function longjmp: 6C56 */
 
-/* function tmpmem_init_dmem: 694A */
+/* function tmpmem_init_dmem: 6969 */
 
 /* function ia_css_dmaproxy_sp_set_addr_B: 3FBD */
 
@@ -47,9 +47,9 @@ more details.
 
 /* function ia_css_queue_get_size: 55CE */
 
-/* function ia_css_queue_load: 5BD5 */
+/* function ia_css_queue_load: 5BF4 */
 
-/* function setjmp: 6C40 */
+/* function setjmp: 6C5F */
 
 /* function ia_css_pipeline_sp_sfi_get_current_frame: 27A0 */
 
@@ -65,9 +65,9 @@ more details.
 #define HIVE_ADDR_sp_sem_for_sp2host_isys_event_queue 0x5804
 #define HIVE_SIZE_sp_sem_for_sp2host_isys_event_queue 20
 
-/* function ia_css_dmaproxy_sp_wait_for_ack: 7223 */
+/* function ia_css_dmaproxy_sp_wait_for_ack: 7242 */
 
-/* function ia_css_sp_rawcopy_func: 5D45 */
+/* function ia_css_sp_rawcopy_func: 5D64 */
 
 /* function ia_css_tagger_buf_sp_pop_marked: 3654 */
 
@@ -107,9 +107,9 @@ more details.
 #define HIVE_ADDR_sp_vbuf_raw 0x39C
 #define HIVE_SIZE_sp_vbuf_raw 4
 
-/* function ia_css_sp_bin_copy_func: 5D26 */
+/* function ia_css_sp_bin_copy_func: 5D45 */
 
-/* function ia_css_queue_item_store: 5933 */
+/* function ia_css_queue_item_store: 5952 */
 
 /* function input_system_reset: 1201 */
 
@@ -209,7 +209,7 @@ more details.
 #define HIVE_ADDR_sp_vbuf_handles 0x7458
 #define HIVE_SIZE_sp_vbuf_handles 960
 
-/* function ia_css_queue_store: 5A99 */
+/* function ia_css_queue_store: 5AB8 */
 
 /* function ia_css_sp_flash_register: 3889 */
 
@@ -281,7 +281,7 @@ more details.
 #define HIVE_ADDR_sp_host_sp_com 0x3E74
 #define HIVE_SIZE_sp_host_sp_com 220
 
-/* function ia_css_queue_get_free_space: 56F8 */
+/* function ia_css_queue_get_free_space: 56FA */
 
 /* function exec_image_pipe: 57A */
 
@@ -297,7 +297,7 @@ more details.
 #define HIVE_ADDR_sp_sp_init_dmem_data 0x73FC
 #define HIVE_SIZE_sp_sp_init_dmem_data 24
 
-/* function ia_css_sp_metadata_start: 60DF */
+/* function ia_css_sp_metadata_start: 60FE */
 
 /* function ia_css_bufq_sp_init_buffer_queues: 38DA */
 
@@ -349,7 +349,7 @@ more details.
 #define HIVE_ADDR_sp_sp_per_frame_data 0x3F50
 #define HIVE_SIZE_sp_sp_per_frame_data 4
 
-/* function ia_css_rmgr_sp_vbuf_dequeue: 669E */
+/* function ia_css_rmgr_sp_vbuf_dequeue: 66BD */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_host2sp_psys_event_queue_handle
@@ -379,7 +379,7 @@ more details.
 
 /* function ia_css_pipeline_sp_run: 1A61 */
 
-/* function memcpy: 6CE0 */
+/* function memcpy: 6CFF */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_N_ISYS2401_DMA_CHANNEL_PROCS
@@ -419,11 +419,11 @@ more details.
 
 /* function stream2mmio_send_command: E04 */
 
-/* function ia_css_uds_sp_scale_params: 69E9 */
+/* function ia_css_uds_sp_scale_params: 6A08 */
 
 /* function ia_css_circbuf_increase_size: 1452 */
 
-/* function __divu: 6C5E */
+/* function __divu: 6C7D */
 
 /* function ia_css_thread_sp_get_state: 1295 */
 
@@ -467,9 +467,9 @@ more details.
 
 /* function ia_css_parambuf_sp_handle_parameter_sets: 18B5 */
 
-/* function ia_css_spctrl_sp_set_state: 60FB */
+/* function ia_css_spctrl_sp_set_state: 611A */
 
-/* function ia_css_thread_sem_sp_signal: 6F44 */
+/* function ia_css_thread_sem_sp_signal: 6F63 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_IRQ_BASE
@@ -483,7 +483,7 @@ more details.
 #define HIVE_ADDR_sp_IRQ_BASE 0x2C
 #define HIVE_SIZE_sp_IRQ_BASE 16
 
-/* function ia_css_virtual_isys_sp_isr_init: 619C */
+/* function ia_css_virtual_isys_sp_isr_init: 61BB */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_TIMED_CTRL_BASE
@@ -497,11 +497,11 @@ more details.
 #define HIVE_ADDR_sp_TIMED_CTRL_BASE 0x40
 #define HIVE_SIZE_sp_TIMED_CTRL_BASE 4
 
-/* function ia_css_isys_sp_generate_exp_id: 652E */
+/* function ia_css_isys_sp_generate_exp_id: 654D */
 
-/* function ia_css_rmgr_sp_init: 6599 */
+/* function ia_css_rmgr_sp_init: 65B8 */
 
-/* function ia_css_thread_sem_sp_init: 7013 */
+/* function ia_css_thread_sem_sp_init: 7032 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_reading_cb_frame
@@ -531,7 +531,7 @@ more details.
 
 /* function csi_rx_backend_rst: CE0 */
 
-/* function ia_css_queue_is_empty: 7370 */
+/* function ia_css_queue_is_empty: 738F */
 
 /* function ia_css_pipeline_sp_has_stopped: 1FC0 */
 
@@ -551,13 +551,13 @@ more details.
 #define HIVE_ADDR_sp_current_sp_thread 0x274
 #define HIVE_SIZE_sp_current_sp_thread 4
 
-/* function ia_css_spctrl_sp_get_spid: 6102 */
+/* function ia_css_spctrl_sp_get_spid: 6121 */
 
 /* function ia_css_bufq_sp_reset_buffers: 3961 */
 
-/* function ia_css_dmaproxy_sp_read_byte_addr: 7251 */
+/* function ia_css_dmaproxy_sp_read_byte_addr: 7270 */
 
-/* function ia_css_rmgr_sp_uninit: 6592 */
+/* function ia_css_rmgr_sp_uninit: 65B1 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_threads_stack
@@ -637,7 +637,7 @@ more details.
 #define HIVE_ADDR_sp_sem_for_reading_cb_param 0x5890
 #define HIVE_SIZE_sp_sem_for_reading_cb_param 40
 
-/* function ia_css_queue_get_used_space: 56AC */
+/* function ia_css_queue_get_used_space: 56AE */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_cont_capt_start
@@ -663,7 +663,7 @@ more details.
 #define HIVE_ADDR_sp_tmp_heap 0x7160
 #define HIVE_SIZE_sp_tmp_heap 640
 
-/* function ia_css_rmgr_sp_get_num_vbuf: 68A2 */
+/* function ia_css_rmgr_sp_get_num_vbuf: 68C1 */
 
 /* function ia_css_ispctrl_sp_output_compute_dma_info: 4C26 */
 
@@ -681,7 +681,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_s3a_bufs 0x5CD8
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_s3a_bufs 60
 
-/* function ia_css_queue_is_full: 5743 */
+/* function ia_css_queue_is_full: 5745 */
 
 /* function debug_buffer_init_isp: E4 */
 
@@ -699,7 +699,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_rmgr_sp_mipi_frame_sem 0x7818
 #define HIVE_SIZE_sp_ia_css_rmgr_sp_mipi_frame_sem 60
 
-/* function ia_css_rmgr_sp_refcount_dump: 6679 */
+/* function ia_css_rmgr_sp_refcount_dump: 6698 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_pipe_private_isp_parameters_id
@@ -741,7 +741,7 @@ more details.
 #define HIVE_ADDR_sp_host2sp_isys_event_queue_handle 0x5D28
 #define HIVE_SIZE_sp_host2sp_isys_event_queue_handle 12
 
-/* function ia_css_thread_sp_yield: 6EC2 */
+/* function ia_css_thread_sp_yield: 6EE1 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_all_cbs_param_desc
@@ -791,9 +791,9 @@ more details.
 
 /* function ia_css_thread_sp_init: 134E */
 
-/* function qos_scheduler_update_fps: 69D9 */
+/* function qos_scheduler_update_fps: 69F8 */
 
-/* function ia_css_isys_sp_reset_exp_id: 6525 */
+/* function ia_css_isys_sp_reset_exp_id: 6544 */
 
 /* function ia_css_ispctrl_sp_set_stream_base_addr: 531A */
 
@@ -827,7 +827,7 @@ more details.
 
 /* function ia_css_dmaproxy_sp_read: 3EA9 */
 
-/* function virtual_isys_stream_is_capture_done: 61C0 */
+/* function virtual_isys_stream_is_capture_done: 61DF */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_raw_copy_line_count
@@ -915,9 +915,9 @@ more details.
 
 /* function debug_enqueue_ddr: EE */
 
-/* function ia_css_rmgr_sp_refcount_init_vbuf: 6634 */
+/* function ia_css_rmgr_sp_refcount_init_vbuf: 6653 */
 
-/* function dmaproxy_sp_read_write: 72EF */
+/* function dmaproxy_sp_read_write: 730E */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_dmaproxy_isp_dma_cmd_buffer
@@ -955,13 +955,13 @@ more details.
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_service 0x307C
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_service 4
 
-/* function ia_css_dmaproxy_sp_process: 703B */
+/* function ia_css_dmaproxy_sp_process: 705A */
 
 /* function ia_css_tagger_buf_sp_mark_from_end: 384E */
 
 /* function ia_css_ispctrl_sp_init_cs: 42F2 */
 
-/* function ia_css_spctrl_sp_init: 6110 */
+/* function ia_css_spctrl_sp_init: 612F */
 
 /* function sp_event_proxy_init: 736 */
 
@@ -1075,7 +1075,7 @@ more details.
 
 /* function pixelgen_tpg_is_done: F07 */
 
-/* function ia_css_isys_stream_capture_indication: 6303 */
+/* function ia_css_isys_stream_capture_indication: 6322 */
 
 /* function sp_start_isp_entry: 392 */
 #ifndef HIVE_MULTIPLE_PROGRAMS
@@ -1091,13 +1091,13 @@ more details.
 
 /* function ia_css_dmaproxy_sp_channel_acquire: 4144 */
 
-/* function ia_css_rmgr_sp_add_num_vbuf: 687E */
+/* function ia_css_rmgr_sp_add_num_vbuf: 689D */
 
 /* function ia_css_tagger_frame_in_capture_pipe: 28DA */
 
 /* function ibuf_ctrl_config: D85 */
 
-/* function ia_css_isys_stream_stop: 6420 */
+/* function ia_css_isys_stream_stop: 643F */
 
 /* function __ia_css_dmaproxy_sp_wait_for_ack_text: 3DFF */
 
@@ -1133,7 +1133,7 @@ more details.
 
 /* function ia_css_tagger_sp_create: 3486 */
 
-/* function tmpmem_acquire_dmem: 692B */
+/* function tmpmem_acquire_dmem: 694A */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_MMU_BASE
@@ -1187,7 +1187,7 @@ more details.
 
 /* function sp_isys_copy_request: 681 */
 
-/* function ia_css_rmgr_sp_refcount_retain_vbuf: 670E */
+/* function ia_css_rmgr_sp_refcount_retain_vbuf: 672D */
 
 /* function ia_css_thread_sp_set_priority: 12E0 */
 
@@ -1197,7 +1197,7 @@ more details.
 
 /* function pixelgen_tpg_stop: EF5 */
 
-/* function tmpmem_release_dmem: 691A */
+/* function tmpmem_release_dmem: 6939 */
 
 /* function __ia_css_dmaproxy_sp_process_text: 3DA3 */
 
@@ -1209,7 +1209,7 @@ more details.
 
 /* function ia_css_tagger_buf_sp_pop_unmarked_and_unlocked: 3608 */
 
-/* function __modu: 6CA4 */
+/* function __modu: 6CC3 */
 
 /* function ia_css_dmaproxy_sp_init_isp_vector: 3E05 */
 
@@ -1231,11 +1231,11 @@ more details.
 #define HIVE_ADDR_sp_GDC_BASE 0x44
 #define HIVE_SIZE_sp_GDC_BASE 8
 
-/* function ia_css_queue_local_init: 590D */
+/* function ia_css_queue_local_init: 592C */
 
-/* function sp_event_proxy_callout_func: 6D71 */
+/* function sp_event_proxy_callout_func: 6D90 */
 
-/* function qos_scheduler_schedule_stage: 6985 */
+/* function qos_scheduler_schedule_stage: 69A4 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_thread_sp_num_ready_threads
@@ -1263,13 +1263,13 @@ more details.
 
 /* function ia_css_ispctrl_sp_isp_done_row_striping: 4C0C */
 
-/* function __ia_css_virtual_isys_sp_isr_text: 617A */
+/* function __ia_css_virtual_isys_sp_isr_text: 6199 */
 
-/* function ia_css_queue_dequeue: 578B */
+/* function ia_css_queue_dequeue: 5792 */
 
-/* function is_qos_standalone_mode: 6960 */
+/* function is_qos_standalone_mode: 697F */
 
-/* function ia_css_dmaproxy_sp_configure_channel: 7268 */
+/* function ia_css_dmaproxy_sp_configure_channel: 7287 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_current_thread_fiber_sp
@@ -1285,7 +1285,7 @@ more details.
 
 /* function ia_css_circbuf_pop: 15EA */
 
-/* function memset: 6D23 */
+/* function memset: 6D42 */
 
 /* function irq_raise_set_token: B6 */
 
@@ -1355,17 +1355,17 @@ more details.
 
 /* function __ia_css_dmaproxy_sp_configure_channel_text: 3F74 */
 
-/* function ia_css_framebuf_sp_wait_for_in_frame: 68A9 */
+/* function ia_css_framebuf_sp_wait_for_in_frame: 68C8 */
 
-/* function ia_css_isys_stream_open: 64D5 */
+/* function ia_css_isys_stream_open: 64F4 */
 
-/* function ia_css_sp_rawcopy_tag_frame: 6055 */
+/* function ia_css_sp_rawcopy_tag_frame: 6074 */
 
 /* function input_system_channel_configure: 11D8 */
 
 /* function isp_hmem_clear: B2D */
 
-/* function ia_css_framebuf_sp_release_in_frame: 68EC */
+/* function ia_css_framebuf_sp_release_in_frame: 690B */
 
 /* function stream2mmio_config: E15 */
 
@@ -1399,7 +1399,7 @@ more details.
 #define HIVE_ADDR_sp_isp_pool 0x390
 #define HIVE_SIZE_sp_isp_pool 4
 
-/* function ia_css_rmgr_sp_rel_gen: 65DB */
+/* function ia_css_rmgr_sp_rel_gen: 65FA */
 
 /* function ia_css_tagger_sp_unblock_clients: 32F2 */
 
@@ -1423,9 +1423,9 @@ more details.
 
 /* function debug_enqueue_isp: 27B */
 
-/* function qos_scheduler_update_stage_budget: 6968 */
+/* function qos_scheduler_update_stage_budget: 6987 */
 
-/* function ia_css_spctrl_sp_uninit: 6109 */
+/* function ia_css_spctrl_sp_uninit: 6128 */
 
 /* function csi_rx_backend_run: C62 */
 
@@ -1475,7 +1475,7 @@ more details.
 #define HIVE_ADDR_sp_ISP_VAMEM_BASE 0x14
 #define HIVE_SIZE_sp_ISP_VAMEM_BASE 12
 
-/* function input_system_channel_sync: 6E3C */
+/* function input_system_channel_sync: 6E5B */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_rawcopy_sp_tagger
@@ -1501,9 +1501,9 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_exp_ids 0x6AE0
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_exp_ids 70
 
-/* function ia_css_queue_item_load: 59FF */
+/* function ia_css_queue_item_load: 5A1E */
 
-/* function ia_css_spctrl_sp_get_state: 60F4 */
+/* function ia_css_spctrl_sp_get_state: 6113 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_callout_sp_thread
@@ -1543,21 +1543,21 @@ more details.
 #define HIVE_ADDR_sp_sp_isp_input_stream_format 0x3E58
 #define HIVE_SIZE_sp_sp_isp_input_stream_format 20
 
-/* function __mod: 6C90 */
+/* function __mod: 6CAF */
 
 /* function ia_css_dmaproxy_sp_init_dmem_channel: 3EC3 */
 
 /* function ia_css_thread_sp_join: 1311 */
 
-/* function ia_css_dmaproxy_sp_add_command: 735A */
+/* function ia_css_dmaproxy_sp_add_command: 7379 */
 
-/* function ia_css_sp_metadata_thread_func: 60ED */
+/* function ia_css_sp_metadata_thread_func: 610C */
 
-/* function __sp_event_proxy_func_critical: 6D5E */
+/* function __sp_event_proxy_func_critical: 6D7D */
 
-/* function ia_css_pipeline_sp_wait_for_isys_stream_N: 62A0 */
+/* function ia_css_pipeline_sp_wait_for_isys_stream_N: 62BF */
 
-/* function ia_css_sp_metadata_wait: 60E6 */
+/* function ia_css_sp_metadata_wait: 6105 */
 
 /* function ia_css_circbuf_peek_from_start: 151A */
 
@@ -1591,7 +1591,7 @@ more details.
 #define HIVE_ADDR_sp_N_CSI_RX_FE_CTRL_DLANES 0x1C4
 #define HIVE_SIZE_sp_N_CSI_RX_FE_CTRL_DLANES 12
 
-/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 723A */
+/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 7259 */
 
 /* function ia_css_ispctrl_sp_done_ds: 4465 */
 
@@ -1637,7 +1637,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_frames 0x6B28
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_frames 280
 
-/* function qos_scheduler_init_stage_budget: 69C6 */
+/* function qos_scheduler_init_stage_budget: 69E5 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp2host_buffer_queue_handle
@@ -1653,11 +1653,11 @@ more details.
 
 /* function ia_css_ispctrl_sp_init_isp_vars: 517F */
 
-/* function ia_css_isys_stream_start: 63B3 */
+/* function ia_css_isys_stream_start: 63D2 */
 
 /* function sp_warning: 8E8 */
 
-/* function ia_css_rmgr_sp_vbuf_enqueue: 66CE */
+/* function ia_css_rmgr_sp_vbuf_enqueue: 66ED */
 
 /* function ia_css_tagger_sp_tag_exp_id: 2A88 */
 
@@ -1665,7 +1665,7 @@ more details.
 
 /* function ia_css_dmaproxy_sp_write: 3E79 */
 
-/* function ia_css_isys_stream_start_async: 647C */
+/* function ia_css_isys_stream_start_async: 649B */
 
 /* function ia_css_parambuf_sp_release_in_param: 187B */
 
@@ -1693,7 +1693,7 @@ more details.
 #define HIVE_ADDR_sp_sp_isp_addresses 0x7094
 #define HIVE_SIZE_sp_sp_isp_addresses 172
 
-/* function ia_css_rmgr_sp_acq_gen: 65F3 */
+/* function ia_css_rmgr_sp_acq_gen: 6612 */
 
 /* function input_system_input_port_open: 10E7 */
 
@@ -1721,7 +1721,7 @@ more details.
 #define HIVE_ADDR_sp_host_sp_queues_initialized 0x3E6C
 #define HIVE_SIZE_sp_host_sp_queues_initialized 4
 
-/* function ia_css_queue_uninit: 58CB */
+/* function ia_css_queue_uninit: 58EA */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_ispctrl_sp_isp_started
@@ -1757,7 +1757,7 @@ more details.
 #define HIVE_ADDR_sp_vbuf_spref 0x394
 #define HIVE_SIZE_sp_vbuf_spref 4
 
-/* function ia_css_queue_enqueue: 5815 */
+/* function ia_css_queue_enqueue: 5828 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_flash_sp_request
@@ -1801,11 +1801,11 @@ more details.
 
 /* function ia_css_pipeline_sp_start: 1FD2 */
 
-/* function ia_css_thread_default_callout: 6EBB */
+/* function ia_css_thread_default_callout: 6EDA */
 
 /* function csi_rx_backend_enable: C3F */
 
-/* function ia_css_sp_rawcopy_init: 5D2D */
+/* function ia_css_sp_rawcopy_init: 5D4C */
 
 /* function input_system_input_port_configure: 1139 */
 
@@ -1863,7 +1863,7 @@ more details.
 #define HIVE_ADDR_sp_sp_all_cbs_frame 0x595C
 #define HIVE_SIZE_sp_sp_all_cbs_frame 16
 
-/* function ia_css_virtual_isys_sp_isr: 739A */
+/* function ia_css_virtual_isys_sp_isr: 73B9 */
 
 /* function thread_sp_queue_print: 13A1 */
 
@@ -1905,9 +1905,9 @@ more details.
 
 /* function ia_css_tagger_buf_sp_pop_from_start: 358A */
 
-/* function __div: 6C48 */
+/* function __div: 6C67 */
 
-/* function ia_css_rmgr_sp_refcount_release_vbuf: 66ED */
+/* function ia_css_rmgr_sp_refcount_release_vbuf: 670C */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_flash_sp_in_use
@@ -1921,7 +1921,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_use 0x5BE8
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_use 4
 
-/* function ia_css_thread_sem_sp_wait: 6F8F */
+/* function ia_css_thread_sem_sp_wait: 6FAE */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_sleep_mode
@@ -1951,7 +1951,7 @@ more details.
 #define HIVE_ADDR_sp_sp_max_cb_elems 0x148
 #define HIVE_SIZE_sp_sp_max_cb_elems 8
 
-/* function ia_css_queue_remote_init: 58ED */
+/* function ia_css_queue_remote_init: 590C */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_isp_stop_req
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_system/spmem_dump.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_system/spmem_dump.c
index 1f78fe0..57bd548 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_system/spmem_dump.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_2401_system/spmem_dump.c
@@ -23,7 +23,7 @@ more details.
 
 /* function input_system_acquisition_stop: AD8 */
 
-/* function longjmp: 6BEF */
+/* function longjmp: 6C0E */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_HIVE_IF_SRST_MASK
@@ -37,9 +37,9 @@ more details.
 #define HIVE_ADDR_sp_HIVE_IF_SRST_MASK 0x1C8
 #define HIVE_SIZE_sp_HIVE_IF_SRST_MASK 16
 
-/* function tmpmem_init_dmem: 6902 */
+/* function tmpmem_init_dmem: 6921 */
 
-/* function ia_css_isys_sp_token_map_receive_ack: 6246 */
+/* function ia_css_isys_sp_token_map_receive_ack: 6265 */
 
 /* function ia_css_dmaproxy_sp_set_addr_B: 3731 */
 
@@ -65,9 +65,9 @@ more details.
 
 /* function ia_css_queue_get_size: 4D4C */
 
-/* function ia_css_queue_load: 5353 */
+/* function ia_css_queue_load: 5372 */
 
-/* function setjmp: 6BF8 */
+/* function setjmp: 6C17 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_sp2host_isys_event_queue
@@ -81,9 +81,9 @@ more details.
 #define HIVE_ADDR_sp_sem_for_sp2host_isys_event_queue 0x46D4
 #define HIVE_SIZE_sp_sem_for_sp2host_isys_event_queue 20
 
-/* function ia_css_dmaproxy_sp_wait_for_ack: 7179 */
+/* function ia_css_dmaproxy_sp_wait_for_ack: 7198 */
 
-/* function ia_css_sp_rawcopy_func: 557D */
+/* function ia_css_sp_rawcopy_func: 559C */
 
 /* function ia_css_tagger_buf_sp_pop_marked: 2DAA */
 
@@ -111,9 +111,9 @@ more details.
 #define HIVE_ADDR_sp_vbuf_raw 0x314
 #define HIVE_SIZE_sp_vbuf_raw 4
 
-/* function ia_css_sp_bin_copy_func: 54A4 */
+/* function ia_css_sp_bin_copy_func: 54C3 */
 
-/* function ia_css_queue_item_store: 50B1 */
+/* function ia_css_queue_item_store: 50D0 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_pipe_private_metadata_bufs
@@ -211,11 +211,11 @@ more details.
 #define HIVE_ADDR_sp_vbuf_handles 0x638C
 #define HIVE_SIZE_sp_vbuf_handles 960
 
-/* function ia_css_queue_store: 5217 */
+/* function ia_css_queue_store: 5236 */
 
 /* function ia_css_sp_flash_register: 2FDF */
 
-/* function ia_css_isys_sp_backend_create: 5EB9 */
+/* function ia_css_isys_sp_backend_create: 5ED8 */
 
 /* function ia_css_pipeline_sp_init: 1896 */
 
@@ -285,7 +285,7 @@ more details.
 #define HIVE_ADDR_sp_host_sp_com 0x413C
 #define HIVE_SIZE_sp_host_sp_com 220
 
-/* function ia_css_queue_get_free_space: 4E76 */
+/* function ia_css_queue_get_free_space: 4E78 */
 
 /* function exec_image_pipe: 658 */
 
@@ -301,7 +301,7 @@ more details.
 #define HIVE_ADDR_sp_sp_init_dmem_data 0x6314
 #define HIVE_SIZE_sp_sp_init_dmem_data 24
 
-/* function ia_css_sp_metadata_start: 5C96 */
+/* function ia_css_sp_metadata_start: 5CB5 */
 
 /* function ia_css_bufq_sp_init_buffer_queues: 304E */
 
@@ -329,7 +329,7 @@ more details.
 #define HIVE_ADDR_sp_sp_per_frame_data 0x4218
 #define HIVE_SIZE_sp_sp_per_frame_data 4
 
-/* function ia_css_rmgr_sp_vbuf_dequeue: 6656 */
+/* function ia_css_rmgr_sp_vbuf_dequeue: 6675 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_host2sp_psys_event_queue_handle
@@ -359,7 +359,7 @@ more details.
 
 /* function ia_css_pipeline_sp_run: 143D */
 
-/* function memcpy: 6C98 */
+/* function memcpy: 6CB7 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_GP_DEVICE_BASE
@@ -387,13 +387,13 @@ more details.
 
 /* function input_system_reg_store: B18 */
 
-/* function ia_css_isys_sp_frontend_start: 60CF */
+/* function ia_css_isys_sp_frontend_start: 60EE */
 
-/* function ia_css_uds_sp_scale_params: 69A1 */
+/* function ia_css_uds_sp_scale_params: 69C0 */
 
 /* function ia_css_circbuf_increase_size: E35 */
 
-/* function __divu: 6C16 */
+/* function __divu: 6C35 */
 
 /* function ia_css_thread_sp_get_state: C78 */
 
@@ -425,9 +425,9 @@ more details.
 
 /* function ia_css_parambuf_sp_handle_parameter_sets: 127F */
 
-/* function ia_css_spctrl_sp_set_state: 5CC5 */
+/* function ia_css_spctrl_sp_set_state: 5CE4 */
 
-/* function ia_css_thread_sem_sp_signal: 6E9A */
+/* function ia_css_thread_sem_sp_signal: 6EB9 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_IRQ_BASE
@@ -453,13 +453,13 @@ more details.
 #define HIVE_ADDR_sp_TIMED_CTRL_BASE 0x40
 #define HIVE_SIZE_sp_TIMED_CTRL_BASE 4
 
-/* function ia_css_isys_sp_isr: 7367 */
+/* function ia_css_isys_sp_isr: 7386 */
 
-/* function ia_css_isys_sp_generate_exp_id: 6467 */
+/* function ia_css_isys_sp_generate_exp_id: 6486 */
 
-/* function ia_css_rmgr_sp_init: 6551 */
+/* function ia_css_rmgr_sp_init: 6570 */
 
-/* function ia_css_thread_sem_sp_init: 6F69 */
+/* function ia_css_thread_sem_sp_init: 6F88 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_reading_cb_frame
@@ -487,7 +487,7 @@ more details.
 
 /* function ia_css_dmaproxy_sp_execute: 3607 */
 
-/* function ia_css_queue_is_empty: 72C6 */
+/* function ia_css_queue_is_empty: 72E5 */
 
 /* function ia_css_pipeline_sp_has_stopped: 186F */
 
@@ -507,13 +507,13 @@ more details.
 #define HIVE_ADDR_sp_current_sp_thread 0x1DC
 #define HIVE_SIZE_sp_current_sp_thread 4
 
-/* function ia_css_spctrl_sp_get_spid: 5CCC */
+/* function ia_css_spctrl_sp_get_spid: 5CEB */
 
 /* function ia_css_bufq_sp_reset_buffers: 30D5 */
 
-/* function ia_css_dmaproxy_sp_read_byte_addr: 71A7 */
+/* function ia_css_dmaproxy_sp_read_byte_addr: 71C6 */
 
-/* function ia_css_rmgr_sp_uninit: 654A */
+/* function ia_css_rmgr_sp_uninit: 6569 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_threads_stack
@@ -531,7 +531,7 @@ more details.
 
 /* function ia_css_parambuf_sp_wait_for_in_param: 1048 */
 
-/* function ia_css_isys_sp_token_map_get_exp_id: 632F */
+/* function ia_css_isys_sp_token_map_get_exp_id: 634E */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_all_cb_elems_param
@@ -583,7 +583,7 @@ more details.
 #define HIVE_ADDR_sp_sem_for_reading_cb_param 0x4760
 #define HIVE_SIZE_sp_sem_for_reading_cb_param 40
 
-/* function ia_css_queue_get_used_space: 4E2A */
+/* function ia_css_queue_get_used_space: 4E2C */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_cont_capt_start
@@ -609,7 +609,7 @@ more details.
 #define HIVE_ADDR_sp_tmp_heap 0x6078
 #define HIVE_SIZE_sp_tmp_heap 640
 
-/* function ia_css_rmgr_sp_get_num_vbuf: 685A */
+/* function ia_css_rmgr_sp_get_num_vbuf: 6879 */
 
 /* function ia_css_ispctrl_sp_output_compute_dma_info: 43A4 */
 
@@ -627,11 +627,11 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_s3a_bufs 0x4BF0
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_s3a_bufs 60
 
-/* function ia_css_queue_is_full: 4EC1 */
+/* function ia_css_queue_is_full: 4EC3 */
 
 /* function debug_buffer_init_isp: E4 */
 
-/* function ia_css_isys_sp_frontend_uninit: 6089 */
+/* function ia_css_isys_sp_frontend_uninit: 60A8 */
 
 /* function ia_css_tagger_sp_exp_id_is_locked: 2084 */
 
@@ -647,7 +647,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_rmgr_sp_mipi_frame_sem 0x674C
 #define HIVE_SIZE_sp_ia_css_rmgr_sp_mipi_frame_sem 60
 
-/* function ia_css_rmgr_sp_refcount_dump: 6631 */
+/* function ia_css_rmgr_sp_refcount_dump: 6650 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_pipe_private_isp_parameters_id
@@ -687,7 +687,7 @@ more details.
 #define HIVE_ADDR_sp_host2sp_isys_event_queue_handle 0x4C40
 #define HIVE_SIZE_sp_host2sp_isys_event_queue_handle 12
 
-/* function ia_css_thread_sp_yield: 6E18 */
+/* function ia_css_thread_sp_yield: 6E37 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_all_cbs_param_desc
@@ -719,7 +719,7 @@ more details.
 
 /* function ia_css_dmaproxy_sp_vmem_read: 35A7 */
 
-/* function ia_css_ifmtr_sp_init: 64B8 */
+/* function ia_css_ifmtr_sp_init: 64D7 */
 
 /* function initialize_sp_group: 668 */
 
@@ -727,9 +727,9 @@ more details.
 
 /* function ia_css_thread_sp_init: D31 */
 
-/* function ia_css_isys_sp_reset_exp_id: 645F */
+/* function ia_css_isys_sp_reset_exp_id: 647E */
 
-/* function qos_scheduler_update_fps: 6991 */
+/* function qos_scheduler_update_fps: 69B0 */
 
 /* function ia_css_ispctrl_sp_set_stream_base_addr: 4A98 */
 
@@ -825,13 +825,13 @@ more details.
 
 /* function encode_and_post_sp_event_non_blocking: A0C */
 
-/* function ia_css_isys_sp_frontend_destroy: 6161 */
+/* function ia_css_isys_sp_frontend_destroy: 6180 */
 
 /* function is_ddr_debug_buffer_full: 2CC */
 
-/* function ia_css_isys_sp_frontend_stop: 60A1 */
+/* function ia_css_isys_sp_frontend_stop: 60C0 */
 
-/* function ia_css_isys_sp_token_map_init: 63FD */
+/* function ia_css_isys_sp_token_map_init: 641C */
 
 /* function ia_css_tagger_buf_sp_get_oldest_marked_offset: 2D1C */
 
@@ -851,9 +851,9 @@ more details.
 
 /* function debug_enqueue_ddr: EE */
 
-/* function ia_css_rmgr_sp_refcount_init_vbuf: 65EC */
+/* function ia_css_rmgr_sp_refcount_init_vbuf: 660B */
 
-/* function dmaproxy_sp_read_write: 7245 */
+/* function dmaproxy_sp_read_write: 7264 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_dmaproxy_isp_dma_cmd_buffer
@@ -891,17 +891,17 @@ more details.
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_service 0x31A0
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_service 4
 
-/* function ia_css_dmaproxy_sp_process: 6F91 */
+/* function ia_css_dmaproxy_sp_process: 6FB0 */
 
 /* function ia_css_tagger_buf_sp_mark_from_end: 2FA4 */
 
-/* function ia_css_isys_sp_backend_rcv_acquire_ack: 5D6E */
+/* function ia_css_isys_sp_backend_rcv_acquire_ack: 5D8D */
 
-/* function ia_css_isys_sp_backend_pre_acquire_request: 5D84 */
+/* function ia_css_isys_sp_backend_pre_acquire_request: 5DA3 */
 
 /* function ia_css_ispctrl_sp_init_cs: 3A66 */
 
-/* function ia_css_spctrl_sp_init: 5CDA */
+/* function ia_css_spctrl_sp_init: 5CF9 */
 
 /* function sp_event_proxy_init: 6C4 */
 
@@ -967,7 +967,7 @@ more details.
 
 /* function sp_dma_proxy_reset_channels: 388C */
 
-/* function ia_css_isys_sp_backend_acquire: 5E8F */
+/* function ia_css_isys_sp_backend_acquire: 5EAE */
 
 /* function ia_css_tagger_sp_update_size: 2C2E */
 
@@ -1027,11 +1027,11 @@ more details.
 
 /* function ia_css_dmaproxy_sp_channel_acquire: 38B8 */
 
-/* function ia_css_rmgr_sp_add_num_vbuf: 6836 */
+/* function ia_css_rmgr_sp_add_num_vbuf: 6855 */
 
 /* function ia_css_tagger_frame_in_capture_pipe: 2030 */
 
-/* function ia_css_isys_sp_token_map_create: 6446 */
+/* function ia_css_isys_sp_token_map_create: 6465 */
 
 /* function __ia_css_dmaproxy_sp_wait_for_ack_text: 3573 */
 
@@ -1067,7 +1067,7 @@ more details.
 
 /* function ia_css_tagger_sp_create: 2BDC */
 
-/* function tmpmem_acquire_dmem: 68E3 */
+/* function tmpmem_acquire_dmem: 6902 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_MMU_BASE
@@ -1131,17 +1131,17 @@ more details.
 
 /* function sp_isys_copy_request: 6A8 */
 
-/* function ia_css_rmgr_sp_refcount_retain_vbuf: 66C6 */
+/* function ia_css_rmgr_sp_refcount_retain_vbuf: 66E5 */
 
 /* function ia_css_thread_sp_set_priority: CC3 */
 
 /* function sizeof_hmem: BF6 */
 
-/* function tmpmem_release_dmem: 68D2 */
+/* function tmpmem_release_dmem: 68F1 */
 
 /* function cnd_input_system_cfg: 392 */
 
-/* function __ia_css_sp_rawcopy_func_critical: 72F0 */
+/* function __ia_css_sp_rawcopy_func_critical: 730F */
 
 /* function __ia_css_dmaproxy_sp_process_text: 3517 */
 
@@ -1153,7 +1153,7 @@ more details.
 
 /* function ia_css_tagger_buf_sp_pop_unmarked_and_unlocked: 2D5E */
 
-/* function __modu: 6C5C */
+/* function __modu: 6C7B */
 
 /* function ia_css_dmaproxy_sp_init_isp_vector: 3579 */
 
@@ -1173,11 +1173,11 @@ more details.
 #define HIVE_ADDR_sp_GDC_BASE 0x44
 #define HIVE_SIZE_sp_GDC_BASE 8
 
-/* function ia_css_queue_local_init: 508B */
+/* function ia_css_queue_local_init: 50AA */
 
-/* function sp_event_proxy_callout_func: 6D29 */
+/* function sp_event_proxy_callout_func: 6D48 */
 
-/* function qos_scheduler_schedule_stage: 693D */
+/* function qos_scheduler_schedule_stage: 695C */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_thread_sp_num_ready_threads
@@ -1205,13 +1205,13 @@ more details.
 
 /* function ia_css_ispctrl_sp_isp_done_row_striping: 438A */
 
-/* function __ia_css_isys_sp_isr_text: 618B */
+/* function __ia_css_isys_sp_isr_text: 61AA */
 
-/* function ia_css_queue_dequeue: 4F09 */
+/* function ia_css_queue_dequeue: 4F10 */
 
-/* function is_qos_standalone_mode: 6918 */
+/* function is_qos_standalone_mode: 6937 */
 
-/* function ia_css_dmaproxy_sp_configure_channel: 71BE */
+/* function ia_css_dmaproxy_sp_configure_channel: 71DD */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_current_thread_fiber_sp
@@ -1227,7 +1227,7 @@ more details.
 
 /* function ia_css_circbuf_pop: FCD */
 
-/* function memset: 6CDB */
+/* function memset: 6CFA */
 
 /* function irq_raise_set_token: B6 */
 
@@ -1259,7 +1259,7 @@ more details.
 #define HIVE_ADDR_sp_isp_ph 0x6348
 #define HIVE_SIZE_sp_isp_ph 28
 
-/* function ia_css_isys_sp_token_map_flush: 638B */
+/* function ia_css_isys_sp_token_map_flush: 63AA */
 
 /* function ia_css_ispctrl_sp_init_ds: 3BF2 */
 
@@ -1293,17 +1293,17 @@ more details.
 
 /* function __ia_css_dmaproxy_sp_configure_channel_text: 36E8 */
 
-/* function ia_css_framebuf_sp_wait_for_in_frame: 6861 */
+/* function ia_css_framebuf_sp_wait_for_in_frame: 6880 */
 
-/* function ia_css_sp_rawcopy_tag_frame: 59EB */
+/* function ia_css_sp_rawcopy_tag_frame: 5A0A */
 
 /* function isp_hmem_clear: B1F */
 
-/* function ia_css_framebuf_sp_release_in_frame: 68A4 */
+/* function ia_css_framebuf_sp_release_in_frame: 68C3 */
 
-/* function ia_css_isys_sp_backend_snd_acquire_request: 5DE1 */
+/* function ia_css_isys_sp_backend_snd_acquire_request: 5E00 */
 
-/* function ia_css_isys_sp_token_map_is_full: 6212 */
+/* function ia_css_isys_sp_token_map_is_full: 6231 */
 
 /* function input_system_acquisition_run: AF3 */
 
@@ -1335,7 +1335,7 @@ more details.
 #define HIVE_ADDR_sp_isp_pool 0x308
 #define HIVE_SIZE_sp_isp_pool 4
 
-/* function ia_css_rmgr_sp_rel_gen: 6593 */
+/* function ia_css_rmgr_sp_rel_gen: 65B2 */
 
 /* function ia_css_tagger_sp_unblock_clients: 2A48 */
 
@@ -1353,15 +1353,15 @@ more details.
 #define HIVE_ADDR_sp_event_any_pending_mask 0x320
 #define HIVE_SIZE_sp_event_any_pending_mask 8
 
-/* function ia_css_isys_sp_backend_push: 5D98 */
+/* function ia_css_isys_sp_backend_push: 5DB7 */
 
 /* function sh_css_decode_tag_descr: 352 */
 
 /* function debug_enqueue_isp: 27B */
 
-/* function qos_scheduler_update_stage_budget: 6920 */
+/* function qos_scheduler_update_stage_budget: 693F */
 
-/* function ia_css_spctrl_sp_uninit: 5CD3 */
+/* function ia_css_spctrl_sp_uninit: 5CF2 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_HIVE_IF_SWITCH_CODE
@@ -1445,11 +1445,11 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_exp_ids 0x59F8
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_exp_ids 70
 
-/* function ia_css_queue_item_load: 517D */
+/* function ia_css_queue_item_load: 519C */
 
-/* function ia_css_spctrl_sp_get_state: 5CBE */
+/* function ia_css_spctrl_sp_get_state: 5CDD */
 
-/* function ia_css_isys_sp_token_map_uninit: 63A8 */
+/* function ia_css_isys_sp_token_map_uninit: 63C7 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_callout_sp_thread
@@ -1477,7 +1477,7 @@ more details.
 #define HIVE_ADDR_sp_SP_PMEM_BASE 0x0
 #define HIVE_SIZE_sp_SP_PMEM_BASE 4
 
-/* function ia_css_isys_sp_token_map_snd_acquire_req: 6318 */
+/* function ia_css_isys_sp_token_map_snd_acquire_req: 6337 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_isp_input_stream_format
@@ -1491,19 +1491,19 @@ more details.
 #define HIVE_ADDR_sp_sp_isp_input_stream_format 0x4120
 #define HIVE_SIZE_sp_sp_isp_input_stream_format 20
 
-/* function __mod: 6C48 */
+/* function __mod: 6C67 */
 
 /* function ia_css_dmaproxy_sp_init_dmem_channel: 3637 */
 
 /* function ia_css_thread_sp_join: CF4 */
 
-/* function ia_css_dmaproxy_sp_add_command: 72B0 */
+/* function ia_css_dmaproxy_sp_add_command: 72CF */
 
-/* function ia_css_sp_metadata_thread_func: 5B96 */
+/* function ia_css_sp_metadata_thread_func: 5BB5 */
 
-/* function __sp_event_proxy_func_critical: 6D16 */
+/* function __sp_event_proxy_func_critical: 6D35 */
 
-/* function ia_css_sp_metadata_wait: 5C85 */
+/* function ia_css_sp_metadata_wait: 5CA4 */
 
 /* function ia_css_circbuf_peek_from_start: EFD */
 
@@ -1513,9 +1513,9 @@ more details.
 
 /* function sp_isys_copy_func: 68A */
 
-/* function ia_css_isys_sp_backend_flush: 5E01 */
+/* function ia_css_isys_sp_backend_flush: 5E20 */
 
-/* function ia_css_isys_sp_backend_frame_exists: 5D1D */
+/* function ia_css_isys_sp_backend_frame_exists: 5D3C */
 
 /* function ia_css_sp_isp_param_init_isp_memories: 4C30 */
 
@@ -1551,7 +1551,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_isys_sp_frontend_states 0x632C
 #define HIVE_SIZE_sp_ia_css_isys_sp_frontend_states 12
 
-/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 7190 */
+/* function ia_css_dmaproxy_sp_read_byte_addr_mmio: 71AF */
 
 /* function ia_css_ispctrl_sp_done_ds: 3BD9 */
 
@@ -1597,11 +1597,11 @@ more details.
 #define HIVE_ADDR_sp_ia_css_bufq_sp_pipe_private_frames 0x5A40
 #define HIVE_SIZE_sp_ia_css_bufq_sp_pipe_private_frames 280
 
-/* function qos_scheduler_init_stage_budget: 697E */
+/* function qos_scheduler_init_stage_budget: 699D */
 
-/* function ia_css_isys_sp_backend_release: 5E76 */
+/* function ia_css_isys_sp_backend_release: 5E95 */
 
-/* function ia_css_isys_sp_backend_destroy: 5EA0 */
+/* function ia_css_isys_sp_backend_destroy: 5EBF */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp2host_buffer_queue_handle
@@ -1615,15 +1615,15 @@ more details.
 #define HIVE_ADDR_sp_sp2host_buffer_queue_handle 0x5B58
 #define HIVE_SIZE_sp_sp2host_buffer_queue_handle 96
 
-/* function ia_css_isys_sp_token_map_check_mipi_frame_size: 62DC */
+/* function ia_css_isys_sp_token_map_check_mipi_frame_size: 62FB */
 
 /* function ia_css_ispctrl_sp_init_isp_vars: 48FD */
 
-/* function ia_css_isys_sp_frontend_has_empty_mipi_buffer_cb: 5EF2 */
+/* function ia_css_isys_sp_frontend_has_empty_mipi_buffer_cb: 5F11 */
 
 /* function sp_warning: 870 */
 
-/* function ia_css_rmgr_sp_vbuf_enqueue: 6686 */
+/* function ia_css_rmgr_sp_vbuf_enqueue: 66A5 */
 
 /* function ia_css_tagger_sp_tag_exp_id: 21DE */
 
@@ -1655,7 +1655,7 @@ more details.
 #define HIVE_ADDR_sp_sp_isp_addresses 0x5FAC
 #define HIVE_SIZE_sp_sp_isp_addresses 172
 
-/* function ia_css_rmgr_sp_acq_gen: 65AB */
+/* function ia_css_rmgr_sp_acq_gen: 65CA */
 
 /* function receiver_reg_load: ACA */
 
@@ -1683,7 +1683,7 @@ more details.
 #define HIVE_ADDR_sp_host_sp_queues_initialized 0x4134
 #define HIVE_SIZE_sp_host_sp_queues_initialized 4
 
-/* function ia_css_queue_uninit: 5049 */
+/* function ia_css_queue_uninit: 5068 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_ispctrl_sp_isp_started
@@ -1729,7 +1729,7 @@ more details.
 #define HIVE_ADDR_sp_sp_metadata_thread 0x4A00
 #define HIVE_SIZE_sp_sp_metadata_thread 72
 
-/* function ia_css_queue_enqueue: 4F93 */
+/* function ia_css_queue_enqueue: 4FA6 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_flash_sp_request
@@ -1757,7 +1757,7 @@ more details.
 #define HIVE_ADDR_sp_tagger_frames 0x4A50
 #define HIVE_SIZE_sp_tagger_frames 168
 
-/* function ia_css_isys_sp_token_map_snd_capture_req: 633A */
+/* function ia_css_isys_sp_token_map_snd_capture_req: 6359 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sem_for_reading_if
@@ -1775,9 +1775,9 @@ more details.
 
 /* function ia_css_pipeline_sp_start: 1881 */
 
-/* function ia_css_thread_default_callout: 6E11 */
+/* function ia_css_thread_default_callout: 6E30 */
 
-/* function ia_css_sp_rawcopy_init: 5565 */
+/* function ia_css_sp_rawcopy_init: 5584 */
 
 /* function tmr_clock_read: 1412 */
 
@@ -1793,7 +1793,7 @@ more details.
 #define HIVE_ADDR_sp_ISP_BAMEM_BASE 0x318
 #define HIVE_SIZE_sp_ISP_BAMEM_BASE 4
 
-/* function ia_css_isys_sp_frontend_rcv_capture_ack: 5FA1 */
+/* function ia_css_isys_sp_frontend_rcv_capture_ack: 5FC0 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_bufq_sp_sems_for_sp2host_buf_queues
@@ -1859,11 +1859,11 @@ more details.
 
 /* function ia_css_tagger_buf_sp_pop_from_start: 2CE0 */
 
-/* function __div: 6C00 */
+/* function __div: 6C1F */
 
-/* function ia_css_isys_sp_frontend_create: 6172 */
+/* function ia_css_isys_sp_frontend_create: 6191 */
 
-/* function ia_css_rmgr_sp_refcount_release_vbuf: 66A5 */
+/* function ia_css_rmgr_sp_refcount_release_vbuf: 66C4 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_ia_css_flash_sp_in_use
@@ -1877,7 +1877,7 @@ more details.
 #define HIVE_ADDR_sp_ia_css_flash_sp_in_use 0x4B00
 #define HIVE_SIZE_sp_ia_css_flash_sp_in_use 4
 
-/* function ia_css_thread_sem_sp_wait: 6EE5 */
+/* function ia_css_thread_sem_sp_wait: 6F04 */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_sp_sleep_mode
@@ -1907,7 +1907,7 @@ more details.
 #define HIVE_ADDR_sp_sp_max_cb_elems 0x148
 #define HIVE_SIZE_sp_sp_max_cb_elems 8
 
-/* function ia_css_queue_remote_init: 506B */
+/* function ia_css_queue_remote_init: 508A */
 
 #ifndef HIVE_MULTIPLE_PROGRAMS
 #ifndef HIVE_MEM_isp_stop_req
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
index b0caf2a..f958bec 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/css_api_version.h
@@ -34,7 +34,7 @@ The version string has four dot-separated numbers, read left to right:
 
 */
 
-#define CSS_API_VERSION_STRING	"2.1.22.2"
+#define CSS_API_VERSION_STRING	"2.1.22.3"
 
 /*
 Change log
@@ -491,6 +491,9 @@ v2.1.22.1 (2 changes parallel), Provide CSS Interface to select correct raw fram
 v2.1.22.2, Enable TDF API
 - add ia_css_tdf_config struct
 
+v2.1.22.3, RGBIR Remosaic ISP Configuration
+- RGBIR Remosaic ISP Configuration
+
 */
 
 #endif /*__CSS_API_VERSION_H*/
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
index 231d478..ddb1674 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160422_0704 */
+/* Release Version: irci_ecr-master_20160426_2016 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
index 5f425ba..ec8438f 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_types.h
@@ -1,4 +1,4 @@
-/* Release Version: irci_ecr-master_20160422_0704 */
+/* Release Version: irci_ecr-master_20160426_2016 */
 /**
 Support for Intel Camera Imaging ISP subsystem.
 Copyright (c) 2010 - 2015, Intel Corporation.
@@ -611,6 +611,7 @@ struct ia_css_isp_config {
 	struct ia_css_frame                      *output_frame;          /**< Output frame the config is to be applied to (optional) */
 	struct ia_css_2500_input_feeder_config   *input_feeder_2500_config;   /**< Input feeder cropping configuration (getters only) */
 	uint32_t			isp_config_id;	/**< Unique ID to track which config was actually applied to a particular frame */
+	struct ia_css_2500_rgbir_kernel_config	 *rgbir_2500_config;	 /**< Skylake: RGBIR Remosaic */
 };
 
 #endif /* _IA_CSS_TYPES_H */
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
index b0f5ecd..ba23529 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/ia_css_version_data.h
@@ -22,7 +22,7 @@ more details.
 #define __IA_CSS_VERSION_DATA_H
 
 
-#define CSS_VERSION_STRING "REL:20160422_16.5_1127; API:2.1.22.2; GIT:irci____4#2c756517f18b06a2b18b93d480cb4474f0f50db1; SDK:; USER:viedifw; "
+#define CSS_VERSION_STRING "REL:20160426_17.2_2334; API:2.1.22.3; GIT:irci___#093cf6b8da3aeaddc74ea86139cac67cf785433b; SDK:; USER:viedifw; "
 
 
 #endif
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/queue/src/queue.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/queue/src/queue.c
index 2fb32d7..14db1f1 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/queue/src/queue.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/runtime/queue/src/queue.c
@@ -114,6 +114,9 @@ int ia_css_queue_enqueue(
 		if (error != 0)
 			return error;
 
+		if (cb_desc.size == 0)
+			return EINVAL;
+
 		/* b. Operate on the queue */
 		if (ia_css_circbuf_desc_is_full(&cb_desc))
 			return ENOBUFS;
@@ -172,6 +175,9 @@ int ia_css_queue_dequeue(
 		if (error != 0)
 			return error;
 
+		if (cb_desc.size == 0)
+			return EINVAL;
+
 		/* b. Operate on the queue */
 		if (ia_css_circbuf_desc_is_empty(&cb_desc))
 			return ENODATA;
@@ -222,8 +228,10 @@ int ia_css_queue_is_full(
 			return error;
 
 		/* b. Operate on the queue */
-		*is_full = ia_css_circbuf_desc_is_full(&cb_desc);
-		return 0;
+		if (cb_desc.size != 0) {
+			*is_full = ia_css_circbuf_desc_is_full(&cb_desc);
+			return 0;
+		}
 	}
 
 	return EINVAL;
@@ -330,7 +338,7 @@ int ia_css_queue_peek(
 
 		/* Check if offset is valid */
 		num_elems = ia_css_circbuf_desc_get_num_elems(&cb_desc);
-		if (offset > num_elems)
+		if (offset > num_elems || cb_desc.size == 0)
 			return EINVAL;
 
 		offset = OP_std_modadd(cb_desc.start, offset, cb_desc.size);
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
index 383f168..cf1046e 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_firmware.c
@@ -47,10 +47,10 @@ struct fw_param {
 /* Warning: same order as SH_CSS_BINARY_ID_* */
 static struct firmware_header *firmware_header;
 
-/* The string STR(irci_ecr-master_20160422_0704) is a place holder
+/* The string STR(irci_ecr-master_20160426_2016) is a place holder
  * which will be replaced with the actual RELEASE_VERSION
  * during package generation. Please do not modify  */
-static const char *release_version = STR(irci_ecr-master_20160422_0704);
+static const char *release_version = STR(irci_ecr-master_20160426_2016);
 
 #define MAX_FW_REL_VER_NAME	300
 static char FW_rel_ver_name[MAX_FW_REL_VER_NAME] = "---";
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.c
index cb7b45e..6db677a 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.c
@@ -3469,6 +3469,7 @@ sh_css_init_isp_params_from_global(struct ia_css_stream *stream,
 		ia_css_lin_set_default_config(&params->lin_2500_config);
 		ia_css_tnr_set_default_config(&params->tnr_2500_config);
 		ia_css_tnr3_set_default_config(&params->tnr3_config);
+		ia_css_rgbir_set_default_config(&params->rgbir_2500_config);
 #endif
 	}
 	else
diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.h b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.h
index 3a50c68..f9b4369 100644
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.h
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/css/sh_css_params.h
@@ -120,6 +120,7 @@ struct ia_css_isp_parameters {
 	struct ia_css_2500_lin_kernel_config  lin_2500_config;
 	struct ia_css_2500_tnr_kernel_config  tnr_2500_config;
 	struct ia_css_tnr3_kernel_config      tnr3_config;
+	struct ia_css_2500_rgbir_kernel_config rgbir_2500_config;
 #endif
 	struct ia_css_dvs_6axis_config  *pipe_dvs_6axis_config[IA_CSS_PIPE_ID_NUM];
 /* ------ deprecated(bz675) : from ------ */
-- 
1.9.1

